### PR TITLE
Added API for reading and writing of floating point numbers

### LINKED ISSUE #167: Missing API: reading and writing of floating point numbers
Issue is present in Okio: https://github.com/square/okio/issues/202, but it is not resolved.

### PR DESCRIPTION
**Pull request by shanshin**
Implemented extension functions for reading and writing values with types Float and Double

Resolves #167

### CODE REVIEW

File: core/common/src/Sinks.kt [hunk 1]
```diff
@@ -275,6 +275,70 @@ public fun Sink.writeULongLe(long: ULong) {
     writeLongLe(long.toLong())
 }
 
+/**
+ * Writes four bytes of a bit representation of [float], in the big-endian order, to this sink.
+ * Bit representation of the [float] corresponds to the IEEE 754 floating-point "single format" bit layout.
+ *
+ * To obtain a bit representation, the [Float.toBits] function is used.
+ *
```
Comment (fzhinkin) [inline core/common/src/Sinks.kt:283]: Does it make sense to mention that these functions should be used with care when working with special values (`NaN` particularly) as bit patterns obtained for `Float.NaN` may vary depending on a platform?

**Review by fzhinkin — APPROVED** [review]
LGTM

**Comment by elect86** [comment]
I'd like this on `BufferedSource` as well

**Comment by fzhinkin** [comment]
@elect86 there's no such entity as `BufferedSource` in `kotlinx-io`, `Source` is already buffered.

**Pull request merged**